src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.js

```diff
@@ -1,28 +1,59 @@
 // @flow
 import { type EdgeCurrencyWallet } from 'edge-core-js'
 
-import { type ApprovableAction, type BorrowEngine, type BorrowRequest, type DepositRequest, type RepayRequest, type WithdrawRequest } from '../../types'
+import {
+  type ApprovableAction,
+  type BorrowCollateral,
+  type BorrowDebt,
+  type BorrowEngine,
+  type BorrowRequest,
+  type DepositRequest,
+  type RepayRequest,
+  type WithdrawRequest
+} from '../../types'
+import { type AaveNetwork } from './AaveNetwork'
 
-export type BorrowEngineBlueprint = {}
+export type BorrowEngineBlueprint = {
+  aaveNetwork: AaveNetwork
+}
 
-export const makeBorrowEngineFactory =
-  (blueprint: BorrowEngineBlueprint) =>
-  async (wallet: EdgeCurrencyWallet): Promise<BorrowEngine> => {
-    return {
-      currencyWallet: wallet,
-      collaterals: [
-        {
-          tokenId: '2260fac5e5542a773aa44fbcfedf7c193bc2c599',
-          nativeAmount: '100000000'
+export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
+  return async (wallet: EdgeCurrencyWallet): Promise<BorrowEngine> => {
+    const { aaveNetwork } = blueprint
+
+    const walletAddress = (await wallet.getReceiveAddress()).publicAddress
+
+    //
+    // Collaterals and Debts
```

Comment:
We have been using `// Collaterals and Debts -------------------------------------------` as our section divider in the core and other parts of the GUI. Just add dashes to get to 80 columns or so.

Reply:
_We agreed to try out the new convention for section headrs_